etna/collections/models.py

```diff
@@ -249,12 +284,38 @@ def related_articles(self):
         from etna.articles.models import ArticlePage
 
         return (
-            ArticlePage.objects.filter(time_period=self)
-            .live()
-            .select_related("teaser_image")
-            .order_by("title")[:3]
+            ArticlePage.objects.live()
+            .public()
+            .filter(id__in=self.related_page_ids)
+            .order_by("-first_published_at")
+            .exclude(title=self.featured_article)
+        )
+
+    @cached_property
+    def related_record_articles(self):
+        from etna.articles.models import RecordArticlePage
+
+        return (
+            RecordArticlePage.objects.live()
+            .public()
+            .filter(id__in=self.related_page_ids)
+            .order_by("-first_published_at")
+            .exclude(title=self.featured_record_article)
```

Comment:
Suggestion: In terms of queries, it's important the order these are made for efficiency. It's better to move the excludes further up the query.

In this query, you're filtering all availble pages by a certain id's, then ordering them, and then excluding. It would be more effecient to exclude first, then filter and etc.